specification/security/resource-manager/Microsoft.Security/preview/2021-01-01-preview/secureScore.json

```diff
@@ -0,0 +1,640 @@
+        "percentage": {
+          "description": "Ratio of the current score divided by the maximum. Rounded to 4 digits after the decimal point",
+          "readOnly": true,
+          "type": "number",
+          "format": "double",
+          "minimum": 0.00,
+          "maximum": 1.00,
+          "exclusiveMinimum": false,
+          "exclusiveMaximum": false
+        }
+      }
+    },
+    "SecureScoreControlList": {
+      "description": "List of security controls",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of security controls in this page",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SecureScoreControlDetails"
+          }
+        },
+        "nextLink": {
+          "description": "The URI to fetch the next page.",
+          "readOnly": true,
+          "type": "string"
+        }
+      }
+    },
+    "SecureScoreControlDetails": {
+      "description": "Details of the security control, its score, and the health status of the relevant resources.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SecureScoreControlScoreDetails"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "SecureScoreControlScoreDetails": {
+      "description": "Calculation result data in control level",
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "description": "User friendly display name of the control",
+          "readOnly": true,
+          "type": "string"
+        },
+        "score": {
+          "description": "Actual score object for the control",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ScoreDetails"
+        },
+        "healthyResourceCount": {
+          "description": "Number of healthy resources in the control",
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32"
+        },
+        "unhealthyResourceCount": {
+          "description": "Number of unhealthy resources in the control",
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32"
+        },
+        "notApplicableResourceCount": {
+          "description": "Number of not applicable resources in the control",
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32"
+        },
+        "weight": {
+          "description": "The relative weight for this specific control in each of your subscriptions. Used when calculating an aggregated score for this control across all of your subscriptions.",
+          "readOnly": true,
+          "type": "integer",
+          "format": "int64",
+          "minimum": 0,
+          "exclusiveMinimum": false
+        },
+        "definition": {
```

Comment:
>definition [](start = 9, length = 10)

I realize this has not changed from the previous api-version but embedding 1 resource within another is a bad-pattern (I realize it is populated only with an expand option) - [https://armwiki.azurewebsites.net/api_contracts/guidelines/resourcegraph.html](https://armwiki.azurewebsites.net/api_contracts/guidelines/resourcegraph.html) - esp. since this is not a child resource but exists at the same or higher scope - the way to do this would be to just have a property 'definitionId' with the value as the full qualified id of the referenced resource

Reply:
@varunkumta since the previous version already GA'ed, won't it be a breaking change ?

Reply:
yes, this is fyi for future changes only

Reply:
please undo this change, as you mentioned it is breaking. If you do want to go this way, only add the definitionId as a new field (do not remove definition)

---
In reply to: [549427772](https://github.com/Azure/azure-rest-api-specs/pull/12181#discussion_r549427772) [](ancestors = 549427772)